src/@ionic-native/plugins/hyper-track/index.ts

```diff
@@ -0,0 +1,246 @@
+ *
+ * // Check if app has location permissions enabled
+ * this.hyperTrack.checkLocationPermission().then(response => {
+ *   // response (String) can be "true" or "false"
+ *   if (response != "true") {
+ *     // Ask for permissions
+ *     this.hyperTrack.requestPermissions().then(response => {}, error => {});
+ *   }
+ * }, error => {});
+ *
+ * // Check if app has location services enabled
+ * this.hyperTrack.checkLocationServices().then(response => {
+ *   // response (String) can be "true" or "false"
+ *   if (response != "true") {
+ *     // Request services to be enabled
+ *     this.hyperTrack.requestLocationServices().then(response => {}, error => {});
+ *   }
+ * }, error => {});
+ *
+ * // First set the current user. This can be done via getOrCreateUser() or setUserId()
+ * this.hyperTrack.setUserId("xxx").then(user => {
+ *   // user (String) is a String representation of a User's JSON
+ *
+ *   this.hyperTrack.startTracking().then(userId => {}, trackingError => {});
+ *
+ *   this.hyperTrack.createAndAssignAction('visit', 'lookupId','address', 20.12, -100.3).then(action => {
+ *     // Handle action. It's a String representation of the Action's JSON. For example:
+ *     // '{"eta":"Jul 17, 2017 12:50:03 PM","assigned_at":"Jul 17, 2017 12:34:38 PM",,"distance":"0.0",...}'
+ *   }, error => {});
+ *
+ *   // You can complete an action with completeAction() or completeActionWithLookupId()
+ *   this.hyperTrack.completeAction('action-id').then(response => {
+ *     // Handle response (String). Should be "OK".
+ *   }, error => {});
+ *
+ *   this.hyperTrack.getCurrentLocation().then(location => {
+ *     // Handle location. It's a String representation of a Location's JSON.For example:
+ *     // '{"mAccuracy":22.601,,"mLatitude":23.123456, "mLongitude":-100.1234567, ...}'
+ *   }, error => {});
+ *
+ *   this.hyperTrack.stopTracking().then(success => {
+ *     // Handle success (String). Should be "OK".
+ *   }, error => {});
+ * 
+ * }, error => {});* 
+ * ```
+ */
+@Plugin({
+  pluginName: 'HyperTrack',
+  plugin: 'cordova-plugin-hypertrack',
+  pluginRef: 'cordova.plugins.HyperTrack',
+  repo: 'https://github.com/hypertrack/hypertrack-cordova',
+  platforms: ['android']
+})
+@Injectable()
+export class HyperTrack extends IonicNativePlugin {
+  /**
+   * Returns given text. For testing purposes.
+   *
+   * @param {String} [text] Given text to print
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with the result text (which is the same as the given text) if successful, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#L6|HyperTrack.js helloWorld implementation}
+   */
+  @Cordova()
+  helloWorld(text: String): Promise<String> { return; }
+
+  /**
+   * Create a new user to identify the current device or get a user from a lookup id.
+   *
+   * @param {String} [name] User's name
```

Comment:
The params are documented as optional, but the method signature defines them as mandatory. Please remove the brackets.